storage/google/cloud/storage/bucket.py

```diff
@@ -748,22 +751,30 @@ def get_blob(
         :rtype: :class:`google.cloud.storage.blob.Blob` or None
         :returns: The blob object if it exists, otherwise None.
         """
+        import time
+        import random
+
         blob = Blob(
             bucket=self,
             name=blob_name,
             encryption_key=encryption_key,
             generation=generation,
             **kwargs
         )
-        try:
-            # NOTE: This will not fail immediately in a batch. However, when
-            #       Batch.finish() is called, the resulting `NotFound` will be
-            #       raised.
-            blob.reload(client=client)
-        except NotFound:
-            return None
-        else:
-            return blob
+        for retry in range(self.RETRIES):  # pragma: no branch
+            try:
+                # NOTE: This will not fail immediately in a batch. However, when
+                #       Batch.finish() is called, the resulting `NotFound` will be
+                #       raised.
+                blob.reload(client=client)
+                break
+            except Exception as e:
+                if retry == self.RETRIES - 1:
+                    raise e
+                else:
+                    rand_num = random.random()
+                    time.sleep(min(rand_num + 2 ** (retry - 1), rand_num + 32))
```

Comment:
@tseaver is there a core library that can be used to replace this inline exponential backoff? gax-go has exponential backoff in its own library.

Could backoff be introduced as a decorator for retryable errors?